### PR TITLE
Benji/jib 137 create app clip unique link

### PR DESCRIPTION
**Pull request by Benjamindodgson**

### CODE REVIEW

File: Shared/API/Onboarding+Cloud.swift [hunk 1]
```diff
@@ -66,15 +68,38 @@ struct VerifyCode: CloudFunction {
...
 struct ActivateUser: CloudFunction {
-    
+
     typealias ReturnType = Any
+    let fullName: String
 
     @discardableResult
     func makeRequest(andUpdate statusables: [Statusable], viewsToIgnore: [UIView]) async throws -> Any {
+
+        let fullName = self.formatName(from: fullName)
```
Comment (Benjamindodgson) [inline Shared/API/Onboarding+Cloud.swift:78]: We now pass the full name in here vs saving to the user before hand. Saves a call.

File: Shared/Flows/Onboarding/OnboardingCoordinator.swift [hunk 2]
```diff
@@ -14,33 +14,20 @@ import Intents
...
-         reservationCreatorId: String?,
-         router: Router,
-         deepLink: DeepLinkable?) {
-
-        self.reservationId = reservationId
-        self.reservationCreatorId = reservationCreatorId ?? "IQgIBSPHpE"
-
-        super.init(router: router, deepLink: deepLink)
-    }
+    lazy var onboardingVC = OnboardingViewController(with: self)
```
Comment (Benjamindodgson) [inline Shared/Flows/Onboarding/OnboardingCoordinator.swift:17]: Just a bunch of cleanup

File: Shared/API/Onboarding+Cloud.swift [hunk 3]
```diff
@@ -66,15 +68,38 @@ struct VerifyCode: CloudFunction {
...
         return try await self.makeRequest(andUpdate: statusables,
-                                               params: [:],
+                                          params: ["givenName": fullName.givenName,
+                                                   "familyName": fullName.familyName],
                                                callName: "setActiveStatus",
                                                delayInterval: 0.0,
                                                viewsToIgnore: viewsToIgnore)
     }
+
+    private func formatName(from text: String) -> (givenName: String, familyName: String) {
```
Comment (MartinJibber) [inline Shared/API/Onboarding+Cloud.swift:88]: Cool

File: Shared/Flows/Onboarding/OnboardingViewController.swift [hunk 4]
```diff
@@ -317,22 +311,35 @@ class OnboardingViewController: SwitchableContentViewController<OnboardingConten
...
     }
 
-    private func handleNameSuccess() {
+    private func handleNameSuccess(for name: String) {
+        self.fullName = name
         // User has been allowed to continue
         if User.current()?.status == .inactive {
             #if APPCLIP
-            self.current = .waitlist(self.waitlistVC)
+            Task {
```
Comment (Benjamindodgson) [inline Shared/Flows/Onboarding/OnboardingViewController.swift:338]: If for some reason we change the flow, we can save the name before we activate the user.

File: Shared/Protocols/Objectable.swift [hunk 5]
```diff
@@ -42,6 +42,21 @@ protocol Objectable: AnyObject {
 
 extension Objectable where Self: PFObject {
 
+    @discardableResult
+    func saveLocally() async throws -> Self {
```
Comment (Benjamindodgson) [inline Shared/Protocols/Objectable.swift:46]: I ended up not using this but its a good function to have.

**Review by MartinJibber — APPROVED** [review]

**Pull request merged**